DataFormats/Luminosity/interface/LumiInfo.h

```diff
@@ -43,6 +46,12 @@ class LumiInfo {
   /// destructor
   ~LumiInfo(){}
 
+  //Total Luminosity Saved 
+  float getTotalLumi() const{return totalLuminosity_;}
```

Comment:
Apologies for not noticing this until now, but this doesn't really make sense. There is already the instLuminosity() method which returns the sum of the individual BX luminosities, which should be the same as this. Really I don't see the purpose of having totalLuminosity_ at all; if we really do want to include this for convenience, it makes the most sense to set it in the constructor when instLumiByBX_ is filled.

Also please name these new variables appropriately. This class is very careful to distinguish between instantaneous luminosity and integrated luminosity and these variables need to be similarly clearly named.

Reply:
Hi Paul.  We added totalLuminosity_ because for PCC the total is not the same as the sum over bxs.  Statistics are limited per BX and so the algo for total is:  total = Sum of all pixel clusters / sum of all events * Nbx (rather than sum over BX of PCC[ibx]/events[ibx]).  So in this case they are not the same and we need the additional float.

What do you prefer for the name of the variable?  totalInstLuminosity_? For the method "getTotalInstLumi()"?

Reply:
Hmm, well, that seems like a recipe for confusion but if it can't be helped, so be it. For the variables, totalInstLuminosity_, totalInstLumiStatErr_, and instLumiStatErrByBX_ seem like the most logical choices (or just Err instead of StatErr if you prefer). As for the methods, the question is what we want to have as the sensible default, so that people won't end up grabbing the wrong method by accident. If you think that the "total" rather than the BX sum is what people are likely to actually want, then I would suggest having just instLuminosity() return the total and rename  BX sum to something less obvious like instLuminosityBXSum(). Then the other methods would be instLuminosityStatErr(), setInstLumiStatErrBX(), setTotalInstLumi(), setTotalInstLumiStatErr(), and getInstLumiStatErrAllBX().

Reply:
Oh, and two more requests:
1) Please make sure that whatever solution you adopt is well-commented, to avoid future confusion!

2) The second constructor also needs to be updated to initialize the total luminosity and statistical error. Actually it might make the most sense to have the second constructor take the total luminosity and the per-BX luminosity (since it doesn't really make sense to set one and not the other) and set the total and per-BX statistical erros to zero, and then add a third constructor which takes the total and per-BX luminosity as well as the total and per-BX statistical error.